Unhandled error during coinjoin api call

The coinjoin completes successfully but it does throw this unhandled error and seems to attempt to broadcast the transaction twice:

```
2021-12-01 09:49:59,504 [INFO]  Transaction seen on network, waiting for confirmation
Unhandled error in Deferred:

Traceback (most recent call last):
  File "/Users/peter/Desktop/Projects/joinmarket-clientserver/jmvenv/lib/python3.9/site-packages/twisted/internet/base.py", line 1318, in run
    self.mainLoop()
  File "/Users/peter/Desktop/Projects/joinmarket-clientserver/jmvenv/lib/python3.9/site-packages/twisted/internet/base.py", line 1328, in mainLoop
    reactorBaseSelf.runUntilCurrent()
  File "/Users/peter/Desktop/Projects/joinmarket-clientserver/jmvenv/lib/python3.9/site-packages/twisted/internet/base.py", line 994, in runUntilCurrent
    call.func(*call.args, **call.kw)
  File "/Users/peter/Desktop/Projects/joinmarket-clientserver/jmvenv/lib/python3.9/site-packages/twisted/internet/task.py", line 253, in __call__
    d = maybeDeferred(self.f, *self.a, **self.kw)
--- <exception caught here> ---
  File "/Users/peter/Desktop/Projects/joinmarket-clientserver/jmvenv/lib/python3.9/site-packages/twisted/internet/defer.py", line 191, in maybeDeferred
    result = f(*args, **kwargs)
  File "/Users/peter/Desktop/Projects/joinmarket-clientserver/jmclient/jmclient/wallet_service.py", line 385, in transaction_monitor
    if f(txd, txid):
  File "/Users/peter/Desktop/Projects/joinmarket-clientserver/jmclient/jmclient/taker.py", line 966, in unconfirm_callback
    self.on_finished_callback(True, fromtx="unconfirmed")
  File "/Users/peter/Desktop/Projects/joinmarket-clientserver/jmclient/jmclient/wallet_rpc.py", line 381, in taker_finished
    assert fromtx is False
builtins.AssertionError:
```

I have a feeling this is related to the above error as now when I try to lock the wallet via the api it fails. Just below the above log output I get the following:
```
<Request at 0x1036e7ac0 method=GET uri=/api/v1/wallet/FullyNoded-qMSDRUxQl1.jmdat/lock clientproto=HTTP/1.1>
b'GET'
b'/api/v1/wallet/FullyNoded-qMSDRUxQl1.jmdat/lock'
{}
b'/api/v1/wallet/FullyNoded-qMSDRUxQl1.jmdat/lock'
<_io.BytesIO object at 0x10380e040>
[(b'Host', [b'censored.onion:28183']), (b'Content-Type', [b'text/plain']), (b'Connection', [b'keep-alive']), (b'Accept', [b'*/*']), (b'User-Agent', [b'FullyNoded/342 CFNetwork/1240.0.4 Darwin/20.6.0']), (b'Authorization', [b'Bearer censored']), (b'Accept-Language', [b'en-us']), (b'Accept-Encoding', [b'gzip, deflate, br'])]

Traceback (most recent call last):
  File "/Users/peter/Desktop/Projects/joinmarket-clientserver/jmvenv/lib/python3.9/site-packages/twisted/web/server.py", line 294, in render
    body = resrc.render(self)
  File "/Users/peter/Desktop/Projects/joinmarket-clientserver/jmvenv/lib/python3.9/site-packages/klein/_resource.py", line 204, in render
    d = defer.maybeDeferred(_execute)
  File "/Users/peter/Desktop/Projects/joinmarket-clientserver/jmvenv/lib/python3.9/site-packages/twisted/internet/defer.py", line 191, in maybeDeferred
    result = f(*args, **kwargs)
  File "/Users/peter/Desktop/Projects/joinmarket-clientserver/jmvenv/lib/python3.9/site-packages/klein/_resource.py", line 196, in _execute
    d = defer.maybeDeferred(
--- <exception caught here> ---
  File "/Users/peter/Desktop/Projects/joinmarket-clientserver/jmvenv/lib/python3.9/site-packages/twisted/internet/defer.py", line 191, in maybeDeferred
    result = f(*args, **kwargs)
  File "/Users/peter/Desktop/Projects/joinmarket-clientserver/jmvenv/lib/python3.9/site-packages/klein/_app.py", line 129, in execute_endpoint
    return endpoint_f(self._instance, *args, **kwargs)
  File "/Users/peter/Desktop/Projects/joinmarket-clientserver/jmvenv/lib/python3.9/site-packages/klein/_app.py", line 241, in _f
    return _call(instance, f, request, *a, **kw)
  File "/Users/peter/Desktop/Projects/joinmarket-clientserver/jmvenv/lib/python3.9/site-packages/klein/_app.py", line 60, in _call
    result = __klein_f__(*args, **kwargs)
  File "/Users/peter/Desktop/Projects/joinmarket-clientserver/jmclient/jmclient/wallet_rpc.py", line 580, in lockwallet
    self.wallet_service.stopService()
  File "/Users/peter/Desktop/Projects/joinmarket-clientserver/jmclient/jmclient/wallet_service.py", line 147, in stopService
    self.monitor_loop.stop()
  File "/Users/peter/Desktop/Projects/joinmarket-clientserver/jmvenv/lib/python3.9/site-packages/twisted/internet/task.py", line 215, in stop
    assert self.running, "Tried to stop a LoopingCall that was " "not running."
builtins.AssertionError: Tried to stop a LoopingCall that was not running.

<Request at 0x1036c2460 method=POST uri=/api/v1/wallet/FullyNoded-qMSDRUxQl1.jmdat/unlock clientproto=HTTP/1.1>
b'POST'
b'/api/v1/wallet/FullyNoded-qMSDRUxQl1.jmdat/unlock'
{}
b'/api/v1/wallet/FullyNoded-qMSDRUxQl1.jmdat/unlock'
<_io.BytesIO object at 0x10383b400>
[(b'Host', [b'censored.onion:28183']), (b'Content-Type', [b'text/plain']), (b'Connection', [b'keep-alive']), (b'Accept', [b'*/*']), (b'User-Agent', [b'FullyNoded/342 CFNetwork/1240.0.4 Darwin/20.6.0']), (b'Content-Length', [b'35']), (b'Accept-Language', [b'en-us']), (b'Accept-Encoding', [b'gzip, deflate, br'])]
2021-12-01 11:09:45,357 [WARNING]  Tried to unlock wallet, but one is already unlocked.
2021-12-01 11:09:45,358 [WARNING]  Currently only one active wallet at a time is supported.
```

--- a/jmclient/jmclient/wallet_rpc.py
+++ b/jmclient/jmclient/wallet_rpc.py
@@ -378,15 +378,22 @@ class JMWalletDaemon(Service):
         # TODO this may be updated.
         # It is also different in that the event loop must not shut down
         # when processing finishes.
-        assert fromtx is False
+
+        # This assertion is making sure that the callback is not a non-final
+        # schedule entry (see the comment above as to why). We accept
+        # either False or 'unconfirmed' since they can arrive in either order.
+        assert fromtx is not True
+
+        # reset our state on completion, we are no longer coinjoining:
+        self.taker = None
+
         if not res:
             jlog.info("Coinjoin did not complete successfully.")
         #Should usually be unreachable, unless conf received out of order;
         #because we should stop on 'unconfirmed' for last (see above)
         else:
             jlog.info("Coinjoin completed correctly")
-        # reset our state on completion, we are no longer coinjoining:
-        self.taker = None
+
         # Note; it's technically possible for this to return False if somehow
         # we are currently in inactive state, but it isn't an error:
         self.activate_coinjoin_state(CJ_NOT_RUNNING)

